guides/source/active_record_validations.md

```diff
@@ -1372,6 +1454,99 @@ irb> person.errors.size
 => 0
 ```
 
+Formatting Full Validation Messages
+-----------------------------------
+
+There are multiple ways to format your error messages.
+
+### i18n
+
+ActiveRecord and ActiveModels both use i18n as their main way to format a
+full validation message. When using i18n you can format your full messages by
+adding a format key to your locale file. For example, if we have a class and
+locale defined as such
+
+```ruby
+class Person < ApplicationRecord
+  validates :name, presence: true
+  validates :age, numericality: true
+end
+```
+
+```yml
+activerecord:
+  errors:
+    models:
+      person:
+        attributes:
+          format: "** %{attribute} %{message} **"
+          name:
+            format: "-- %{message} --"
+            blank: "can't be blank"
+          age:
+            not_a_number: "is not a number"
```

Comment:
This requires `config.active_model.i18n_customize_full_message = true` in application.rb file.